File: ts/problems/58-length-of-last-word.test.ts
import { describe, expect, test } from "vitest";
import { lengthOfLastWord } from "./58-length-of-last-word";

describe("lengthOfLastWord", () => {
  test.each([
    {
      s: "",
      want: 0,
    },
    {
      s: "Hello World",
      want: 5,
    },
    {
      s: "   fly me   to   the moon  ",
      want: 4,
    },
    {
      s: "luffy is still joyboy",
      want: 6,
    },
  ])("given %s we expect %want", ({ s, want }) => {
    expect(lengthOfLastWord(s)).toBe(want);
  });
});

File: go/problems/58_length_of_last_word_test.go
package main

import (
	"fmt"
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestLengthOfLastWord(t *testing.T) {
	tests := []struct {
		s    string
		want int
	}{
		{
			s:    "",
			want: 0,
		},
		{
			s:    "Hello World",
			want: 5,
		},
		{
			s:    "   fly me   to   the moon  ",
			want: 4,
		},
		{
			s:    "luffy is still joyboy",
			want: 6,
		},
	}

	for _, tt := range tests {
		t.Run(fmt.Sprintf("for %q the length of last word is %d", tt.s, tt.want), func(t *testing.T) {
			got := lengthOfLastWord(tt.s)
			assert.Equal(t, tt.want, got, "got %d want %d", got, tt.want)
		})
	}
}
